substrate-node/pallets/pallet-smart-contract/src/tests.rs

```diff
@@ -1113,41 +1082,10 @@ fn test_node_contract_billing_works_for_non_existing_accounts() {
 
         info!("total amount billed {:?}", total_amount_billed);
 
-        let staking_pool_account_balance = Balances::free_balance(&get_staking_pool_account());
-        info!(
-            "staking pool account balance, {:?}",
-            staking_pool_account_balance
-        );
-
-        // 5% is sent to the staking pool account
-        assert_eq!(
-            staking_pool_account_balance,
-            (Perbill::from_percent(5) * total_amount_billed)
-        );
-
-        // 10% is sent to the foundation account
-        let pricing_policy = TfgridModule::pricing_policies(1).unwrap();
-        let foundation_account_balance = Balances::free_balance(&pricing_policy.foundation_account);
-        assert_eq!(
-            foundation_account_balance,
-            (Perbill::from_percent(10) * total_amount_billed)
-        );
-
-        // 50% is sent to the sales account
-        let sales_account_balance = Balances::free_balance(&pricing_policy.certified_sales_account);
-        assert_eq!(
-            sales_account_balance,
-            (Perbill::from_percent(50) * total_amount_billed)
-        );
+        let total_balance = Balances::total_balance(&twin.account_id);
+        let total_amount_billed = initial_twin_balance - total_balance;
```

Comment:
same here

Reply:
This code is used to calculate the initial balance for the source account.
 I believe you are questioning the flag that was just added to the validate_distribution_rewards() function?
 The flag was added to handle the initial balance for destination accounts.

Reply:
Actually I was referring to duplicated calculation of `total_amount_billed`

![image](https://github.com/user-attachments/assets/b8203b2a-1f5d-4083-89cf-f4092cb48a80)

Reply:
My bad :(